kclvm/tools/src/LSP/src/goto_def.rs

```diff
@@ -30,59 +37,212 @@ pub(crate) fn goto_definition(
     match program.pos_to_stmt(kcl_pos) {
         Some(node) => match node.node {
             Stmt::Import(stmt) => goto_def_for_import(&stmt, kcl_pos, prog_scope, program),
-            _ => {
-                let objs = find_definition_objs(node, kcl_pos, prog_scope);
-                let positions = objs
-                    .iter()
-                    .map(|obj| (obj.start.clone(), obj.end.clone()))
-                    .collect();
-                positions_to_goto_def_resp(&positions)
-            }
+            _ => match find_def(node.clone(), kcl_pos, prog_scope) {
+                Some(obj) => {
+                    let mut positions = IndexSet::new();
+                    positions.insert((obj.start.clone(), obj.end.clone()));
+                    return positions_to_goto_def_resp(&positions);
+                }
+                None => return None,
+            },
         },
         None => None,
     }
 }
-pub(crate) fn find_definition_objs(
+
+// Todo: fix ConfigExpr
+// ```kcl
+// schema Person:
+//     name: str
+//     data: Data
+
+// schema Data:
+//     id: int
+
+// person = Person {
+//     data.id = 1
+//     data: {
+//         id = 1
+//     }
+//     data: Data {
+//         id = 3
+//     }
+// }
+
+pub(crate) fn find_def(
     node: Node<Stmt>,
     kcl_pos: &KCLPos,
     prog_scope: &ProgramScope,
-) -> Vec<ScopeObject> {
+) -> Option<ScopeObject> {
     let (inner_expr, parent) = inner_most_expr_in_stmt(&node.node, kcl_pos, None);
     if let Some(expr) = inner_expr {
         if let Expr::Identifier(id) = expr.node {
-            let name = get_identifier_last_name(&id);
-            let objs = if let Some(parent) = parent {
-                // find schema attr def
-                match parent.node {
+            let id_node = Node::node_with_pos(
+                id.clone(),
+                (
+                    expr.filename,
+                    expr.line,
+                    expr.column,
+                    expr.end_line,
+                    expr.end_column,
+                ),
+            );
+            let id = pre_process_identifier(id_node, kcl_pos);
+            match parent {
+                Some(schema_expr_node) => match schema_expr_node.node {
+                    // find_schema_attr
                     Expr::Schema(schema_expr) => {
-                        find_def_of_schema_attr(schema_expr, prog_scope, name)
+                        let schema_obj = find_def(node, &schema_expr.name.get_pos(), prog_scope);
+                        match schema_obj {
+                            Some(schema) => match &schema.ty.kind {
```

Comment:
Use the `if let` pattern instead of `match xxx` Perhaps you need to carefully check the code style, and later we will add stricter lint checks.